test/studies/ml/performance/xor_perceptron.good

```diff
@@ -0,0 +1,6 @@
+xor_perceptron.chpl:95: warning: In an upcoming release 'e' will no longer be included by default, please 'use' or 'import' the 'Math' module to access it
```

Comment:
Probably should be fixed.

Reply:
This must be a mistake by the compiler. The error is referencing
```chapel
for e in 1..epochs {
    tn.shuffle(batch);
    tn.shuffle(batchSizes);
    const loss = train(batch[0..#(batchSizes[0])],learnRate);
}
```

Reply:
`e` is an auto imported symbol from the math module that you are shadowing here which is why you're getting a deprecation warning for it. I'd change that loop to:
```chapel
for 1..epochs { ... }
```
since `e` isn't used anyway.

Reply:
Nonetheless, I agree with Iain that the compiler should have done a better job here. If it is an easy fix to not issue that warning here, I think it'd be worthwhile.

I am especially worried if I'd get warnings with:

```chpl
var e = ...;

proc/iter foo(e) { ...

proc/iter e { ...

proc/iter e() { ...

record/class/module e { ...
```